cub/cub/device/dispatch/kernels/scan.cuh

```diff
@@ -159,7 +161,9 @@ template <typename ChainedPolicyT,
           typename InitValueT,
           typename OffsetT,
           typename AccumT,
-          bool ForceInclusive>
+          bool ForceInclusive,
+          typename RealInitValueT = ::cuda::std::
+            _If<::cuda::std::is_void_v<typename InitValueT::value_type>, InitValueT, typename InitValueT::value_type>>
```

Comment:
important: as we discussed offline, this can be simplified to:
```suggestion
          typename RealInitValueT = typename InitValueT::value_type>
```
C layer always passes `InitValueT` here, so the default is never used, and C++ layer always has `::value_type`. If we wanted to remove last template parameter from the C layer, this would stop working, since the entire overload would be SFINAE-d out.

Reply:
Fixed https://github.com/NVIDIA/cccl/pull/3462/commits/f20219b479501b55066c5ecc2f7767b108fbe53c